records_mover/db/mysql/load_options.py

```diff
@@ -0,0 +1,289 @@
+from sqlalchemy.sql.expression import text, TextClause
+from records_mover.records import RecordsHints
+from records_mover.utils import quiet_remove
+from records_mover.records.hints import cant_handle_hint
+from records_mover.records import DelimitedRecordsFormat
+from records_mover.records.types import (
+    HintEncoding, HintRecordTerminator,
+    HintFieldDelimiter, HintQuoteChar,
+    HintQuoting, HintEscape,
+    HintHeaderRow, HintCompression,
+    HintDoublequote,
+)
```

Comment:
I used this PR to experiment with using MyPy's `Literal` type to represent the various possible values for different hints:

https://mypy.readthedocs.io/en/stable/literal_types.html

As a result, I'm more confident the code below covers all of the legal hints in converting them to MySQL load configuration.

What's not done yet is any real central validation that the hint that e.g. may have been passed by another Python program or read from a file in S3 is actually valid by the spec.  

Now that issue is a little nuanced, as we do want to try to be forward compatible against future expansions to the spec, which is why we have a "fail_if_cant_handle_hint" ProcessingInstruction flag that you can control.

I'm going to try out doing central validation as a PR on this PR and see how it goes.  This would get rid of all of the `type: ignore` comments below at the very least.

Reply:
I'm going to proceed with the PR (https://github.com/bluelabsio/records-mover/pull/50) as a separate task and go ahead and get this change into master in the meantime.